src/EventStore.ClientAPI/Internal/EventStoreConnectionLogicHandler.cs

```diff
@@ -163,18 +166,28 @@ internal class EventStoreConnectionLogicHandler : IEventStoreConnectionLogicHand
 				return;
 			}
 
+			var useSsl = _compatibilityMode.IsAutoCompatibilityModeEnabled()
+				? endPoints.SecureTcpEndPoint != null
+				: _settings.UseSslConnection;
+
 			_connectingPhase = ConnectingPhase.ConnectionEstablishing;
 			_connection = new TcpPackageConnection(
 				_settings.Log,
 				endPoint,
 				Guid.NewGuid(),
-				_settings.UseSslConnection,
+				useSsl,
 				_settings.ValidateServer,
 				_settings.ClientConnectionTimeout,
 				(connection, package) => EnqueueMessage(new HandleTcpPackageMessage(connection, package)),
 				(connection, exc) => EnqueueMessage(new TcpConnectionErrorMessage(connection, exc)),
 				connection => EnqueueMessage(new TcpConnectionEstablishedMessage(connection)),
 				(connection, error) => EnqueueMessage(new TcpConnectionClosedMessage(connection, error)));
+
+			if (_connection.ConnectionId == Guid.Empty) {
```

Comment:
Won't do anything as `TcpPackageConnection` was patched in `v5-master` because the client used to silently crashed if SSL was enable but `HostHeader` was null or empty. Returning an empty `Guid` was used to communicate that the connection was misconfigured in order to have a proper error. I don't see `Ensure.NotNullOrEmpty` used on `TargetHost` on `master`.

Reply:
`v5-master` context: https://github.com/EventStore/EventStore/blob/v5-master/src/EventStore.ClientAPI/Transport.Tcp/TcpPackageConnection.cs#L53